BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m

```diff
@@ -471,9 +471,7 @@ - (NSMutableURLRequest *)requestWithMultipartFormRequest:(NSURLRequest *)request
         [inputStream close];
 
         if (handler) {
-            dispatch_async(dispatch_get_main_queue(), ^{
-                handler(error);
-            });
+            handler(error);
```

Comment:
How was this causing a deadlock? Are we calling `prepareAPIRequest` from the main thread, or is it internal?

Reply:
more details are in the ticket. There's no reason for building multipart file to callback onto main thread internally to transfer to next phase. Also, doing this is not expected by the caller which usually calls prepareOperation synchronously, potentially on the main thread, which calls prepareAPIRequest internally, blocked on a background thread which is blocked on main thread, thus, results in deadlock.